### PR TITLE
Change render function on Render to return Result

### PR DESCRIPTION
**Pull request by synlestidae**
Fixed a portion of the unwraps in render.rs for #3 

Have more changes to make before that is solved :)

### CODE REVIEW

File: src/render.rs [hunk 1]
```diff
@@ -35,28 +35,34 @@ impl ForLoop {
...
-    output: String,
     context: Json,
     ast: Node,
-    for_loops: Vec<ForLoop>
+    for_loops: Vec<ForLoop>,
+    render_error: Option<RenderError>
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub struct RenderError {
```
Comment (Keats) [inline src/render.rs]: can you put the errors in a separate file and maybe use https://github.com/tailhook/quick-error or structure them like https://github.com/kbknapp/clap-rs/blob/master/src/errors.rs ?

File: src/render.rs [hunk 2]
```diff
@@ -275,50 +296,57 @@ impl Renderer {
...
             i += 1;
         }
+        Ok(output)
     }
 
-    pub fn render_node(&mut self, node: Node) {
+    pub fn render_node(&mut self, node: Node) -> Result<String, RenderError> {
         match node.specific {
-            Text(ref s) => self.output.push_str(s),
+            Text(ref s) => Ok(s.to_string()),
```
Comment (Keats) [inline src/render.rs]: `to_owned` is preferrable to `to_string`

File: tests/templates/basic_space.html [hunk 3]
```diff
@@ -0,0 +1 @@
+<h1>{{object.doesnt_exist}}</h1>
```
Comment (Keats) [inline tests/templates/basic_space.html]: name of the file seems wrong?

**Comment by Keats** [comment]
Looks good for a start!
Note i'll be away for 2 weeks without laptop on wednesday so i won't be commenting too much (or at all) / merging during that time

**Comment by synlestidae** [comment]
Okay that's cool! 

I hope that's all the feedback you have for now. I'll implement it later today when I have the project open

**Comment by Keats** [comment]
I had to change the render a bit to add inheritance (see #12) but that shouldn't be a big change

**Comment by Keats** [comment]
@synlestidae do you think you will have time to work on it this week? i'd like to release a 0.1 on crates.io asap.
Otherwise I can continue from your pr

**Comment by Keats** [comment]
Closed in favour of #16

**Pull request closed without merging**